core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1059,16 +1046,16 @@ class DAGScheduler(
 
         if (disallowStageRetryForTest) {
           abortStage(failedStage, "Fetch failure will not retry stage due to testing config")
-        } else if (failedStages.isEmpty && eventProcessActor != null) {
+        } else if (failedStages.isEmpty && eventProcessLoop != null) {
```

Comment:
I checked the history. `eventProcessLoop` was initialized in the `start` method. Since now we initialize it in the constructor, we can remove the null check.